src/libraries/Common/src/System/Security/Cryptography/MLDsaCng.Windows.cs

```diff
@@ -95,16 +95,21 @@ static Exception DebugFailAndGetException(string? parameterSet)
             }
         }
 
-        public partial CngKey Key
+        public partial CngKey GetKey()
         {
-            get
-            {
-                ThrowIfDisposed();
+            ThrowIfDisposed();
 
-                return _key;
-            }
+#if SYSTEM_SECURITY_CRYPTOGRAPHY
+            return CngHelpers.Duplicate(_key.HandleNoDuplicate, _key.IsEphemeral);
+#else
+#pragma warning disable CA1416 // only supported on: 'windows'
```

Comment:
Can we just do a platform check and throw if not supported instead of the pragma?